### PR TITLE
Add the premium api

### PR DESCRIPTION
**Pull request by arnaudlevy**
Hi @wikiti! 
Thanks for the gem :)
I'm trying to add the premium API.
It's not working yet, though.
@sebouchu do you know why my ENV vars are not loaded?
Sorry for the bad coding style, it's not yet clear for me.
Yours,
Arnaud

### CODE REVIEW

**Comment by SebouChu** [comment]
@arnaudlevy Missing the dotenv gem : https://github.com/bkeepers/dotenv

But dotenv should not be a dependency here, it's up to the developer using this gem to make sure the environement variables are loaded

**Comment by arnaudlevy** [comment]
@SebouChu when loaded in Osuny, there's no env vars loaded by the gem

**Comment by wikiti** [comment]
Hey folks! Thanks for taking the time for improving the gem 😄 

I'm not sure if hardcoding the premium feature in this gem is the best approach tho. The gem was initially built as a wrapper for the LanguageTool open source tool, rather than for its hosted/commercial version.

The gem should be compatible with the premium version by simply specifying the base url and some extra query parameters during API calls (`user` and `apiKey`). I think the gem is fully compatible right now with the premium requirements.

However, I think that a minor refactor can be done to the gem to prevent the verbosity of having to add a `user` and a `apiKey` query parameters to every call made. What do you think about having some sort of `default_query_params` to the API initializer? Something like this:

````rb
LanguageTool::API.new(
  base_uri: 'https://api.languagetoolplus.com/v2',
  common_query_params: {
    username: '...',
    apiKey: '...'
  }
)
````

Then, this query parameters would be passed to every request made through the API wrapper. It could also be configured to provide other parameters that might change in the future as well.

**Comment by arnaudlevy** [comment]
Thanks!
In rails there is something like default_params, no?

**Comment by wikiti** [comment]
> Thanks! In rails there is something like default_params, no?

Not sure about rails conventions, but this a ruby gem, not a rails gem 😉  Feel free to use whatever name you please!

**Comment by arnaudlevy** [comment]
Hi @wikiti :) 
What do you think?

File: lib/languagetool/actions/check.rb [hunk 1]
```diff
@@ -15,7 +15,14 @@ def run
...
       def query
-        normalize_query(PARAMETERS.map { |k| [k, options[k]] })
+        normalize_query(parameters_with_common_query_params)
+      end
+
+      def parameters_with_common_query_params
+        PARAMETERS.map { |k| [k, options[k]] } + [
+          [:username, api.username],
+          [:api_key, api.api_key]
+        ]
```
Comment (wikiti) [inline lib/languagetool/actions/check.rb]: These changes should belong to the base action `LanguageTool::Actions::Base`. You also shouldn't hardcode which parameters are sent, but let the coder specify them via `common_query_params`.

I think `normalize_query` can hold this logic, and you could simply do something like this:

````rb
def normalize_query(query)
  query = api.common_query_params.merge(query)

  # ...
end
````

**Review by wikiti — COMMENTED** [review]
Looking good overall! I think you shouldn't specify which common attributes to use, and simply let the coder choose which ones are to be sent to the API 😄 

Thanks!

**Comment by SebouChu** [comment]
Hello @wikiti! I updated the branch so that no hardcoded query param name is in the code.

I also removed the `default_query` method which was not used.

Tell me if you need something else before merging ;)

**Pull request merged**